### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by ZewdieMc**
## Hey Reviewer :wave: 
In this PR, we implemented the following[ project requirements.](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md)
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements
- Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. `rentals.json`)
- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+    # book["rentals"] << self
+    # person.rentals << self
```
Comment (youmari) [inline classes/rental.rb]: - [ ] it would be better to remove the commented code in order to keep the file clean and pro

File: .rubocop.yml [hunk 2]
```diff
@@ -18,7 +18,12 @@ Metrics/ClassLength:
 Metrics/BlockLength:
   IgnoredMethods: ['describe']
   Max: 30
-
+Metrics/CyclomaticComplexity:
+  Max: 10
+Metrics/ParameterLists:
+  Enabled: false
+Style/OptionalBooleanParameter:
+  Enabled: false
```
Comment (youmari) [inline .rubocop.yml]: - [ ] You shouldn't disable robocop parameters because it ensures you write clean and pro code

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @ZewdieMc @David-Lanzz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @ZewdieMc @David-Lanzz , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Preserve data works as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**